spec/acceptance/replset_spec.rb

```diff
@@ -202,16 +200,7 @@ class { 'mongodb::globals':
         nc1ohyB0lNt8lHf1U00mtgDSV3fwo5LkwhRi6d+bDBTL/C6MZETMLdyCqDlTdUWG
         YXIsJ0gYcu9XG3mx10LbdPJvxSMg',
         }
-        -> class { 'mongodb::client': }
-      EOS
-
-      apply_manifest_on(hosts.reverse, pp, catch_failures: true)
-      apply_manifest_on(hosts.reverse, pp, catch_changes: true)
-    end
-
-    it 'sets up the replset with puppet' do
-      pp = <<~EOS
-        mongodb_replset { 'test':
+        -> mongodb_replset { 'test':
```

Comment:
Tried to move `mongodb_replset` and `mongodb_user` into the first puppet run on both nodes.
This way when mongodb_replset is realized on the fist node it strips away the second node ( because it's inactive at the time ) before invoking rs.initiate().

This almost worked, but then the idepotency run on host 2 fails because since host 2 is not master, mongodb_user doesn't fetch users.

```
  	Warning: User info is available only from master host
  	Warning: User creation is available only from master host
  	�[mNotice: /Stage[main]/Main/Mongodb_user[User admin on db admin]/ensure: created
```